setup/setup_native.sh

```diff
@@ -0,0 +1,141 @@
+echo "Ensure we exit on error"
+set -e
+
+source setup/setup_shared.sh
+
+export PLATFORMS=""
+
+if [ $SETUP_IOS ] || ( [ !$SETUP_ANDROID ] && [ !$SETUP_IOS ] ); then
+    OSX_MAJOR_VERSION=`sw_vers | grep ProductVersion | cut -d ':' -f 2 | cut -d '.' -f 1`
+    echo "Found OSX major version" $OSX_MAJOR_VERSION
+
+    # The Homebrew pac-man is installed in different locations, depending on whether the processor
+    # is an Intel or Apple Silicone chip.  Intel uses x86_64, Apple chips are amd64, so we can
+    # check the chip type using these hardware platforms.
+    CHIP_ARC=`uname -m`
+    INTEL="x86_64"
+    APPLE_SILICONE="arm64"
+    WORKING_DIR=""
+
+    if [ $CHIP_ARC == $INTEL ]; then
+        echo "Found "$INTEL" chip"
+        WORKING_DIR="/usr/local/"
+    else
+        if [ $CHIP_ARC == $APPLE_SILICONE ]; then
+            echo "Found "$APPLE_SILICONE" chip"
+            WORKING_DIR=$HOMEBREW_PREFIX
+        fi
+    fi
+
+    CURR_RUBY_VERSION=`ruby --version | cut -d ' ' -f 2 | cut -d '.' -f 1-2`
+    echo "Found ruby version "$CURR_RUBY_VERSION
+
+    if [ $CURR_RUBY_VERSION == $RUBY_VERSION ]; then
+        echo "Found ruby version "$CURR_RUBY_VERSION" expected "$RUBY_VERSION" no need to upgrade"
+    else
+        if [ -x "${WORKING_DIR}/bin/brew" ]; then
+            echo "Found brew installation with version" ` brew --version`
+            echo "Installing ruby version to brew" $RUBY_VERSION
+            brew install ruby@$RUBY_VERSION
+        else
+            if [ $OSX_MAJOR_VERSION -ge $OSX_EXP_VERSION ]; then
+                echo "No brew installation found, but OSX major version "$OSX_MAJOR_VERSION" and expected version "$OSX_EXP_VERSION" so CocoaPods should work"
```

Comment:
nit; future fix: I think we should remove this check. Per @iantei, it doesn't actually work on recent versions of android. We should just say that brew is a requirement.